neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltPreparedStatement.java

```diff
@@ -118,7 +118,7 @@ private StatementResult executeInternal() throws SQLException {
 	}
 
 	private boolean hasResultSet() {
-		return this.statement != null && this.statement.toLowerCase().contains("return");
+		return this.statement != null && this.statement.toLowerCase().contains(" return ");
```

Comment:
@jexp Much better, I can open an issue for that.
I'd love to remove as much as possible `String.contains` from the jdbc code, relying on the inner driver. Thinking that HTTP implementation could be a problem, as it cannot benefit of that APIs...